### PR TITLE
DO catalog map

### PR DESCRIPTION
**Pull request by Jesus89**
https://app.clubhouse.io/cartoteam/epic/107513/dataset-visualisation-in-do-catalogue

### CODE REVIEW

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+  computed: {
+    ...mapState({
+      dataset: state => state.catalog.dataset,
+      variables: state => state.catalog.variables
+    }),
+    title () {
+      return this.dataset.name;
+    },
+    categoryId () {
+      console.log('CATEGORY ID:', this.dataset.category_id);
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:93]: I guess we don't want that log anymore...

Comment (Jesus89) [reply lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:93]: Yep, I'll review all the logs. Thanks

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+    },
+    variableMax () {
+      return this.formatNumber(this.variable && this.variable.max);
+    },
+    variableAvg () {
+      return this.formatNumber(this.variable && this.variable.avg);
+    },
+    variableBins () {
+      if (this.variable && this.variable.quantiles && colorStyle) {
+        const breaks = [...this.variable.quantiles[2]['5'], this.variable.max];
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:119]: Why 2 and '5' ? Maybe we could add some constants for those 'magic numbers'...

Comment (Jesus89) [reply lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:119]: Yep, this is a bit ugly, but it's how quantiles are packaged.

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+    });
+
+    this.renderLayer();
+  },
+  methods: {
+    importMapboxStyles () {
+      let style = document.createElement('link');
+      style.type = 'text/css';
+      style.rel = 'stylesheet';
+      style.href = 'https://api.mapbox.com/mapbox-gl-js/v1.12.0/mapbox-gl.css';
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:222]: We're in the middle of a change to replace the external CDN use of mapbox-gl-js code, so this should be now replaced with: `https://libs.cartocdn.com/mapbox-gl/v1.13.0/mapbox-gl.css`

This is our own copy of latest BSD mapbox (v.1.13.0), but in our CDN

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+          },
+          getFillColor,
+          getLineColor,
+          getLineWidth,
+          getRadius,
+          lineWidthUnits: 'pixels',
+          pointRadiusUnits: 'pixels',
+          pickable: true,
+          onDataLoad: (tileJSON) => {
+            console.log('TILEJSON', tileJSON);
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue]: console.log!

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+            const { center, tilestats } = tileJSON;
+            this.initialViewState = {
+              zoom: parseFloat(center[2]) - 1,
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:264]: Why the use of -1 here? Did we create that config refered to gmaps or anything similar?

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+      });
+      if (variable.quantiles && variable.quantiles.length === undefined) {
+        // Compatibility change: convert JSON into array
+        variable.quantiles = Object.keys(variable.quantiles).map(k => ({ [k]: variable.quantiles[k] }));
+      }
+      this.variable = {
+        ...variable,
+        ...variableExtra
+      };
+      console.log('VARIABLE', { ...this.variable });
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue]: log

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+        getFillColor = [234, 200, 100, 168];
+        getLineColor = [44, 44, 44, 60];
+        getLineWidth = 1;
+      } else if (g === 'Polygon' && v === 'Number') {
+        colorStyle = colorBinsStyle({
+          breaks: { stats, method: 'quantiles', bins: 5 },
+          colors: this.categoryIdPalette
+        });
+        getFillColor = (d) => colorStyle(d.properties[propId]);
+        getLineColor = [44, 44, 44, 60];
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:332]: It looks like some common style props could be extracted and reused

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+      return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
+    },
+    generateColorStyle () {
+      const g = this.geomType;
+      const v = this.variable && this.variable.type;
+      const stats = this.variable;
+
+      propId = this.variable && this.variable.attribute;
+
+      if (g === 'Polygon' && this.isGeography) {
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:322]: The module is quite long, so this styling logic could be probably extracted to an autonomous module, for easier testing.

File: lib/assets/javascripts/new-dashboard/components/Catalog/legends/BasicLegend.vue [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+export default {
+  name: 'BasicLegend',
+  props: {
+    title: String
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+// .basic-legend {}
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/legends/BasicLegend.vue:17]: not used...

File: lib/assets/javascripts/new-dashboard/components/Catalog/map-styles/colorBinsStyle.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import { range } from 'd3-array';
+import { scaleThreshold } from 'd3-scale';
+import { gePalette, NULL_COLOR } from './utils';
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/Catalog/map-styles/colorBinsStyle.js:3]: is `gePalette` a typo for `getPalette`?

File: lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue [hunk 11]
```diff
@@ -45,13 +45,21 @@
           :message="$t('FeedbackMessage.message')"
           @click.native.stop.prevent="toggleDropdown"/>
 
-        <NotificationPopup
+        <!-- <NotificationPopup
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue:48]: should we remove this commented block?

File: lib/assets/javascripts/new-dashboard/pages/Data/CatalogDatasetMap.vue [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+      return this.dataset.name;
+    },
+    defaultSource () {
+      return this.dataset.sample_info && this.dataset.sample_info.default_source;
+    },
+    hasSample () {
+      return this.dataset.sample_info && !!this.dataset.sample_info.id;
+    }
+  },
+  methods: {
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/pages/Data/CatalogDatasetMap.vue:52]: empty methods

File: package.json [hunk 13]
```diff
@@ -39,15 +44,18 @@
...
     "d3-queue": "^3.0.7",
+    "d3-scale": "^3.2.3",
     "date-fns": "^1.29.0",
     "fastclick": "^1.0.6",
     "html-webpack-plugin": "^3.2.0",
     "internal-carto.js": "CartoDB/carto.js#v4.2.2-1",
     "jquery": "2.1.4",
     "leaflet": "CartoDB/Leaflet#v1.3.1-carto1",
     "loader-utils": "https://registry.npmjs.org/loader-utils/-/loader-utils-0.2.17.tgz",
+    "mapbox-gl": "^1.12.0",
```
Comment (VictorVelarde) [inline package.json:58]: I guess ^1.13.0 should be the default, the latest BSD one

**Review by VictorVelarde — APPROVED** [review]
LGTM. Nice job with the deck.gl integration, legends and so on!
Just added some comments, most of them related to possible small refactors.

File: lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue (hunk 5 above)
Comment (Jesus89) [reply lib/assets/javascripts/new-dashboard/components/Catalog/CatalogMap.vue:264]: Product wanted to zoom out the zoom value provided by the tiler. I'll add a comment in the code

**Comment by Jesus89** [comment]
Hey have refactored the colorStyles functions and added tests! :D

Also included some last minute changes:
- Add catalog map attribution
- Move default map warning to the top

**Pull request merged**